ignore/ignore.go

```diff
@@ -0,0 +1,73 @@
+// package ignore provides helpers to support ignore-files similar to .gitignore
+package ignore
+
+import (
+	"bufio"
+	"io"
+	"strings"
+)
+
+var (
+	LineComment = "#"
+	IgnoreFile  = ".sourcegraph/sourcegraphignore"
+)
+
+type IgnoreMatcher struct {
+	IgnoreList []string
+	Strip      int
```

Comment:
do we still need strip now that we won't support archive?

Reply:
I checked, searcher won't require ` strip` either. Let's remove it.